pkg/cmd/prerunner.go

```diff
@@ -538,6 +538,12 @@ func (r *PreRun) createMDSClient(ctx *config.Context, ver *version.Version, unsa
 // Initializes a default KafkaRestClient
 func (r *PreRun) InitializeOnPremKafkaRest(command *AuthenticatedCLICommand) func(*cobra.Command, []string) error {
 	return func(cmd *cobra.Command, args []string) error {
+		// ErrIfMissingRunRequirement is called inside of AuthenticatedWithMDS but we don't return that error
+		// So call it in this func too to avoid skipping the annotation check
```

Comment:
Any way we can only call it once / why don't we return the error from AuthenticatedWithMDS?

Reply:
We don't return the error because `InitializeOnPremKafkaRest` uses the existence of an error to set the bool `useMdsToken := err == nil`; and it prompts the user for username/password if `useMdsToken` is false.

Instead of calling `ErrIfMissingRunRequirement` twice, we could instead check if the error from `AuthenticatedWithMDS` contains text from the `ErrIfMissingRunRequirement` error. If that sounds good I can go ahead and make that change.

Reply:
If the code won't be too messy, let's go ahead with that!

Reply:
> contains text from the ErrIfMissingRunRequirement error

I think the more idiomatic way to do this would be to define a new "RunRequirementError" type and check to see if the error is of that type.